pydra/engine/core.py

```diff
@@ -566,24 +615,84 @@ def split(self, splitter, overwrite=False, cont_dim=None, **kwargs):
         if cont_dim:
             for key, vel in cont_dim.items():
                 self._cont_dim[f"{self.name}.{key}"] = vel
-        if kwargs:
-            self.inputs = attr.evolve(self.inputs, **kwargs)
+        if inputs:
+            new_inputs = {}
+            split_inputs = set(
+                f"{self.name}.{n}" if "." not in n else n
+                for n in self._unwrap_splitter(splitter)
+                if not n.startswith("_")
+            )
+            for inpt_name, inpt_val in inputs.items():
```

Comment:
Just a comment, but I really feel uncomfortable with weird variable name shortcuts such as `fld`, `mdata` or `inpt`. IMO, it adds unnecessary cognitive load to save a limited amount of keystrokes.

Reply:
Yes, it isn't my normal style, I just tried to keep it as close to the original as possible